arches_component_lab/utils/safe_object_filter.py

```diff
@@ -0,0 +1,22 @@
+import logging
+from django.core.exceptions import FieldError
+
+logger = logging.getLogger(__name__)
+
+
+class SafeObjectFilter(object):
+
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+
+    def apply_filters(self, **kwargs):
+        object_filter = self.model.objects
+        for arg in kwargs.items():
+            try:
+                object_filter = object_filter.filter(arg)
+            except FieldError:
```

Comment:
I'm worried this will hide legitimate programming mistakes. I frequently have haste-makes-waste scenarios where a FieldError from django helps me catch `node__cardmodel` when I meant `node__nodegroup__cardmodel` etc. I would be a little sad to lose that.

But this is just component lab and only used in a couple of places, so I don't feel too strongly.

Reply:
This occurred to me too, and I don't love it but I thought it might be better than to have it spread across a number of classes. If the general consensus is to have the version-specific decision logic in the python code I'll drop what I've done here.